pkg/k8/deployments/deployments.go

```diff
@@ -2,37 +2,105 @@ package deployments
 
 import (
 	"fmt"
-	"strings"
+	"os"
 	"time"
 
+	"github.com/okteto/cnd/pkg/k8/util"
+	"github.com/okteto/cnd/pkg/model"
 	log "github.com/sirupsen/logrus"
 
 	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8Yaml "k8s.io/apimachinery/pkg/util/yaml"
 	"k8s.io/client-go/kubernetes"
 )
 
-//Deploy deploys a k8 deployment
-func Deploy(d *appsv1.Deployment, namespace string, c *kubernetes.Clientset) error {
-	deploymentName := GetFullName(namespace, d.Name)
-	dClient := c.AppsV1().Deployments(namespace)
-	dk8, err := dClient.Get(d.Name, metav1.GetOptions{})
-	if err != nil && !strings.Contains(err.Error(), "not found") {
-		return fmt.Errorf("Error getting kubernetes deployment: %s", err)
+const (
+	deploymentFile = "deployment.yml"
+)
+
+//DevDeploy deploys a k8 deployment in dev mode
+func DevDeploy(dev *model.Dev, namespace string, c *kubernetes.Clientset) (string, error) {
+	d, err := loadDeployment(dev, namespace, c)
+	if err != nil {
+		return "", err
 	}
 
-	if dk8.Name == "" {
+	parentRevision := d.GetObjectMeta().GetAnnotations()[model.RevisionAnnotation]
+	if d.GetObjectMeta().GetLabels()[model.CNDLabel] != "" {
+		log.Debugf("The current deployment %s is already in dev mode. Leaving the original parent revision.", GetFullName(d.Namespace, d.Name))
+		parentRevision = d.GetObjectMeta().GetAnnotations()[model.CNDRevisionAnnotation]
+	}
+
+	dev.TurnIntoDevDeployment(d, parentRevision)
+
+	name, err := deploy(d, c)
+	if err != nil {
+		return "", err
+	}
+
+	return name, nil
+}
+
+//Deploy deploys a k8 deployment in prod mode
+func Deploy(dev *model.Dev, namespace string, c *kubernetes.Clientset) (string, error) {
+	if dev.Swap.Deployment.File != "" {
```

Comment:
this is to handle back compat. Should we just remove it and add a message to our users to do a down and up ?